src/pybrake/middleware/fastapi.py

```diff
@@ -0,0 +1,193 @@
+        metric.end_time = time.time()
+        notifier.routes.notify(metric)
+        set_active_metrics(None)
+
+
+def _before_cursor(notifier):
+    def _sqla_before_cursor_execute(
+            conn, cursor, statement, parameters, context, executemany
+    ):
+        if notifier and not notifier.config.get("performance_stats"):
+            return
+        start_span("sql")
+
+    return _sqla_before_cursor_execute
+
+
+def _after_cursor(notifier):
+    def _sqla_after_cursor_execute(
+            conn, cursor, statement, parameters, context, executemany
+    ):
+        if notifier and not notifier.config.get("performance_stats"):
+            return
+        end_span("sql")
+        metric = get_active_metrics()
+        if metric is not None:
+            notifier.queries.notify(
+                query=statement,
+                method=getattr(metric, "method", ""),
+                route=getattr(metric, "route", ""),
+                start_time=metric.start_time,
+                end_time=time.time(),
+            )
+
+    return _sqla_after_cursor_execute
+
+
+def init_app(app: FastAPI, sqlEngine: future.Engine = None) -> FastAPI:
+    """
+    Initiate PyBrake notifier and apply patch for Error monitoring and APM.
+    :param app: instance of FastAPI application
+    :param sqlEngine: SQLALCHEMY engine instance
+    :return: FastAPI application instance after apply patch or new setting
+    """
+    if "pybrake" in app.extra:
+        raise ValueError("pybrake is already injected")
+    if "PYBRAKE" not in app.extra:
+        raise ValueError("app.config['PYBRAKE'] is not defined")
+
+    @app.middleware("http")
+    async def init_request_vars(request: Request, call_next):
+        current_request.set(request)
+        response = await call_next(request)
+        return response
+
+    notifier = Notifier(**app.extra["PYBRAKE"])
+    app.extra["pybrake"] = notifier
+    config = notifier.config
+
+    notifier.add_filter(request_filter)
+
+    # Error notification patch
+    old_exception_error_call = ExceptionMiddleware.__call__
+
+    async def patch_exception_error_call(
+            self, scope: Scope, receive: Receive, send: Send) -> None:
+        try:
+            await old_exception_error_call(self, scope, receive, send)
+        except Exception as exc:
+            notifier.notify(exc)
+            raise exc
+
+    ExceptionMiddleware.__call__ = patch_exception_error_call
+
+    # Route Stats monitoring
+    @app.middleware("http")
+    async def process_route_stats(request: Request, call_next):
+        if config.get("performance_stats"):
+            _before_request(request)
+        response = await call_next(request)
+        if config.get("performance_stats"):
+            _after_request(response, notifier)
+        return response
+
+    # Route breakdown stats monitoring
+    old_render = Response.render
+
+    def patch_render(self, content) -> bytes:
```

Comment:
Why do we annotate the return type? We didn't do that in previous PRs.

Reply:
Actually, I added a patch here and created a new method in the same way as method define in framework. This is also good practise. So...